hatebase/__init__.py

```diff
@@ -1,37 +1,222 @@
+        """getAnalysis gets the analysis results of a previously performed analyze query.
+
+        (From hatebase-api documentation: he /get_analysis endpoint retrieves the assessment
+        initiated with a prior query to the /analyze endpoint. Analysis can take anywhere
+        from a few minutes to a couple hours, depending on the complexity of the content
+        and the volume of traffic at the time of request.
+        Requests auto-expire within 48 hours
+
+        Args:
+            filters (dict):    dictionary of filters for the get_analysis query. (see hatebase
+                documentation for details). Must contain the "request_id".
+            format:     format of the response (currently only json)
+
+        Returns:
+            response:   the full json response according to the hatebase api.
+        """
+
+        url = self.base_url + '/' + self.version + '/get_analysis'
+        payload = "token=" + self.token + "&format=" + format + "&" + self.format_query(filters)
+        if self.debug == True:
+            print("url: {}\npayload: {}".format(url, payload))
+        # perform query
+        response = requests.post(url, data=payload, headers=self.headers)
+        if self.debug == True:
+            print("response: {}".format(response))
+        resp_json = response.json()
+        return resp_json
+
+    def getVocabulary(self, filters, format="json"):
+        """get the desired vocabulary / vocabulary entries from hatebase.
+
+        The /get_vocabulary endpoint allows users to download Hatebase's
+        lexicon of multilingual hate speech.
+        Note that resultsets are paginated and that vocabulary is frequently
+        updated based on usage, particularly sightings.
+
+        Args:
+            filters (dict):     dictionary of filters for the get_vocabulary query. (see hatebase
+                documentation for details).
+            format:     format of the response (currently only json)
+
+        Returns:
+            response:   the full json response according to the hatebase api.
+            vocabulary: a list of vocabulary entries (each entry is a dict) for the current page
+            pages:      the total number of result_pages for this query
+            results:    the total numvber of results (vocabulary entries) for this query
+            language:   the language of the vocabulary
+        """
+
+        url = self.base_url + '/' + self.version + '/get_vocabulary'
+        payload = "token=" + self.token + "&format=" + format + "&" + self.format_query(filters)
+        if self.debug == True:
+            print("url: {}\npayload: {}".format(url, payload))
+        # perform query
+        response = requests.post(url, data=payload, headers=self.headers)
+        if self.debug == True:
+            print("response: {}".format(response))
+        resp_json = response.json()
+        return resp_json, resp_json["result"], resp_json["number_of_pages"], \
```

Comment:
I was wondering if you could go into more detail why you chose to return a tuple here instead of the resp_json?  I'm open to your thoughts! :-)

I'm initially of the mind that it's easier to just return everything in one dictionary, so people don't have to remember the order that things are returned.

Reply:
My hatebase queries so far were almost always about getting the full vocabulary of a given language (mostly english & german)... so I always got the total number of results & number of pages from the first response as a first step. This is why I figured it would be nice to return these values directly.

But I do agree that this is a different behavior compared to all other replies. So maybe it's better to always just return the full response... or to also find meaningful return parameters for the other functions so that always a tuple is returned containing the full response at first position and the other values thereafter.

But I can totally live with either way / decision. What do you prefer?

Reply:
Hi. I apologize for the late reply. Please have them return the resp_json. Thank you :-)

Reply:
done ;-)